### PR TITLE
Bound values added in fields of ObservedQuery and ObservedBatch struct

### PR DESCRIPTION
**Pull request by tushardas**
The statements passed in the statement fields of ObservedQuery and ObservedBatch, do not account for the bound values passed while executing a query. 

Refer issue: #1497 for more details

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -1995,6 +2001,7 @@ type QueryObserver interface {
 type ObservedBatch struct {
 	Keyspace   string
 	Statements []string
+	Values     [][]interface{}
```
Comment (martin-sucha) [inline session.go:2004]: Could you please add documentation for the field?

For example:

```go
// Values holds a slice of bound values for each statement.
// Values[i] are bound values passed to Statements[i].
// Do not modify the values here, they are shared with multiple goroutines.
Values     [][]interface{}
```

File: session.go [hunk 2]
```diff
@@ -1957,6 +1962,7 @@ func (t *traceWriter) Trace(traceId []byte) {
 type ObservedQuery struct {
 	Keyspace  string
 	Statement string
+	Values    []interface{}
```
Comment (martin-sucha) [inline session.go:1965]: Could you please add documentation for the field?

For example:

```go
// Values holds a slice of bound values for the query.
// Do not modify the values here, they are shared with multiple goroutines.
Values     [][]interface{}
```

**Review by martin-sucha — CHANGES_REQUESTED** [review]

File: session.go (hunk 2 above)
Comment (tushardas) [reply session.go]: @martin-sucha Documentation is added. Thanks

File: session.go (hunk 1 above)
Comment (tushardas) [reply session.go]: Documentation is added. Thanks

**Review by martin-sucha — APPROVED** [review]

**Comment by tushardas** [comment]
@Zariel @jeffwidman Can you please look into this PR?

**Comment by jeffwidman** [comment]
@tushardas I'm not a maintainer so @martin-sucha would be the right person to chat with.

Also FYI there's a merge conflict you might want to first resolve.

File: cassandra_test.go [hunk 3]
```diff
@@ -1907,6 +1909,9 @@ func TestBatchObserve(t *testing.T) {
 		if stmt != fmt.Sprintf(`INSERT INTO batch_observe_table (id,other) VALUES (?,%d)`, i) {
 			t.Fatal("unexpected query", stmt)
 		}
+		if observedBatch.observedValues[i] != []interface{}{i} {
```
Comment (martin-sucha) [inline cassandra_test.go:1912]:
```
./cassandra_test.go:1912:38: invalid operation: observedBatch.observedValues[i] != []interface {} literal (slice can only be compared to nil)
./cassandra_test.go:1913:81: observedBatch.observedEntries undefined (type *observation has no field or method observedEntries)
```

`observedBatch.observedEntries` should be `observedBatch.observedValues` instead. But both lines could be replaced with a call to `assertDeepEqual(t, "observed value", []interface{}{i}, observedBatch.observedValues[i]`

https://github.com/gocql/gocql/blob/87a5d7a5ff74ffa9eb716f05f6e26191836bf03d/common_test.go#L263-L268

**Review by martin-sucha — COMMENTED** [review]
Thanks for resolving the merge conflict! There is a build failure in tests that we need to resolve before merging (see inline comment).

File: cassandra_test.go (hunk 3 above)
Comment (tushardas) [reply cassandra_test.go]: @martin-sucha Fixed the tests. Thanks for suggesting the fix

**Comment by martin-sucha** [comment]
Merged. Thanks!

**Pull request merged**